scripting/functions/listeners/convarChange.sp

```diff
@@ -0,0 +1,40 @@
+public void OnConVarChange_checkSurrender(ConVar convar, char[] oldValue, char[] newValue) {
+    if (RealPlayerCount(0, false, false, false) <= 0) return;
+
+    int CTScore = CS_GetTeamScore(CS_TEAM_CT);
+    int TScore = CS_GetTeamScore(CS_TEAM_T);
+    int totalScore = CTScore + TScore;
+    int maxRounds = GetConVarInt(FindConVar("mp_maxrounds"))
+
+    if (totalScore >= maxRounds) {
+        totalScore = totalScore - maxRounds;
+        maxRounds = GetConVarInt(FindConVar("mp_overtime_maxrounds"));
+
+        while (totalScore >= maxRounds) {
+            totalScore = totalScore - maxRounds;
+        }
+    }
+
+    int res = RoundToNearest(float(maxRounds / 2)) - totalScore;
+
+    if (res == 1) { // Match is now 1 round before halftime
+        canSurrender = false;
+    }
+    if (res == ((RoundToNearest(float(maxRounds / 2)) - maxRounds) + 1)) {
+        canSurrender = false;
+    }
+    // TODO: Check players if game is 5v3 if yes then surrender is available.
```

Comment:
Are you going to do this?

Reply:
This will be to come back to later on if players need this or report it as an issue.